WMF Framework/ReadingListsSyncOperation.swift

```diff
@@ -752,14 +752,14 @@ internal class ReadingListsSyncOperation: ReadingListsOperation {
                 guard !isDeleted else {
                     return
                 }
-                guard let articleURL = remoteEntry.articleURL, let articleKey = articleURL.wmf_inMemoryKey else {
+                guard let articleURL = remoteEntry.articleURL, let articleKey = articleURL.wmf_inMemoryKey, let remoteArticleKey = remoteEntry.articleKey else {
                     return
                 }
                 remoteEntriesToCreateLocallyByArticleKey[articleKey] = remoteEntry
-                guard !requestedArticleKeys.contains(articleKey) else {
+                guard !requestedArticleKeys.contains(remoteArticleKey) else {
                     return
                 }
-                requestedArticleKeys.insert(articleKey)
+                requestedArticleKeys.insert(remoteArticleKey)
                 if let article = dataStore.fetchArticle(with: articleURL, in: moc) {
                     articlesByKey[articleKey] = article
                 } else {
```

Comment:
I'm seeing a crash at [line 822](https://github.com/wikimedia/wikipedia-ios/blob/VariantAwareSaveButtonsAndList/WMF%20Framework/ReadingListsSyncOperation.swift#L822) when I log in with an account with reading lists and it begins syncing. It's expecting `articleKey` to be of type `NSString`. That `moc` call seems pretty untyped (it accepts `Any` for value), maybe we can replace it with one of our article-specific `fetchOrCreate` calls like `moc.fetchOrCreateArticle(withKey: articleKey.databaseKey, variant: articleKey.languageVariantCode)`?

Reply:
I'll take a look at both and thanks for finding those.

Reply:
The crash is a line I missed in putting that change into a commit - and the fix is exactly as you suggested.

Will see what is up with the save button selection and hopefully have both fixes in a small commit.